.buildkite/pipeline.yml

```diff
@@ -284,6 +284,34 @@ steps:
                   branch: "${BUILDKITE_BRANCH}"
                   env:
                     - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
+                    - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+
+  - label: "Trigger x-pack/libbeat"
+    plugins:
+      - monorepo-diff#v1.0.1:
+          diff: "git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD"
+          watch:
+            - path:
+                - x-pack/libbeat/
+                - .buildkite/x-pack/pipeline.xpack.libbeat.yml
+                - .buildkite/scripts
+                - .buildkite/hooks
+                #OSS
+                - go.mod
+                - pytest.ini
+                - dev-tools/
+                - testing/
+                # x-pack
+                - libbeat/
```

Comment:
My understanding from [/Jenkinsfile.yml](https://github.com/elastic/beats/blob/main/Jenkinsfile.yml#L27-L39) is that the common changesets for x-pack are:

```
 # x-pack
                - go.mod
                - pytest.ini
                - dev-tools/
                - libbeat/
                - testing/
                - x-pack/libbeat/
```
we've already used this pattern in [x-pack/heartbeat](https://github.com/elastic/beats/blob/main/.buildkite/pipeline.yml#L326-L332) so perhaps we can standardize i.e. replace both oss and x-pack with one "# x-pack" section for x-pack related projects (similar to [Jenkinsfile](https://github.com/elastic/beats/blob/main/x-pack/libbeat/Jenkinsfile.yml#L6))?

Reply:
ok, let's compare what we have:
```
                **- x-pack/libbeat/**
                - .buildkite/x-pack/pipeline.xpack.libbeat.yml
                - .buildkite/scripts
                - .buildkite/hooks
                #OSS
                **- go.mod**
                **- pytest.ini**
                **- dev-tools/**
                **- testing/**
                # x-pack
                **- libbeat/**
```

Reply:
or do you mean we shouldn't merge them?

Reply:
they are the same, I am just saying that for consistency's sake we should have a section

```
# x-pack
 # x-pack
                - go.mod
                - pytest.ini
                - dev-tools/
                - libbeat/
                - testing/
                - x-pack/libbeat/
```

for all x-pack pipelines. The pipeline specific bits will, of course, remain above, so the total thing with my proposal would be:

```
- x-pack/<whatever beat>/ # exceptionally here, since it's x-pack/libbeat which is part of the the x-pack common section below we should skip this line
- .buildkite/x-pack/pipeline.xpack.libbeat.yml
- .buildkite/scripts/
- .buildkite/hooks/
# x-pack
- go.mod
- pytest.ini
- dev-tools/
- libbeat/
- testing/
- x-pack/libbeat/
```